zc_common/remote_resource/request.py

```diff
@@ -0,0 +1,84 @@
+import requests
+from inflection import underscore
+from uritemplate import expand, variables
+from zc_common.jwt_auth.utils import service_jwt_payload_handler, jwt_encode_handler
+from zc_common.settings import zc_settings
+
+
+class RemoteResourceWrapper(object):
+
+    def __init__(self, data):
+        self.data = data
+        self.create_properties_from_data()
+
+    def create_properties_from_data(self):
+        if 'id' in self.data:
+            setattr(self, 'id', self.data.get('id'))
+
+        if 'type' in self.data:
+            setattr(self, 'type', self.data.get('type'))
+
+        if 'attributes' in self.data:
+            attributes = self.data['attributes']
+            for key in attributes.keys():
+                setattr(self, underscore(key), attributes[key])
+
+        if 'relationships' in self.data:
+            relationships = self.data['relationships']
+
+            for key in relationships.keys():
+                if isinstance(relationships[key]['data'], list):
+                    setattr(self, underscore(key), RemoteResourceListWrapper(relationships[key]['data']))
+                else:
+                    setattr(self, underscore(key), RemoteResourceWrapper(relationships[key]['data']))
+
+
+class RemoteResourceListWrapper(list):
+    def __init__(self, data):
+        self.data = data
+        self.add_items_from_data()
+
+    def add_items_from_data(self):
+        map(lambda x: self.append(RemoteResourceWrapper(x)), self.data)
+
+
+def get_route_from_fk(resource_type, url_param_values={}):
+    """Gets a fully qualified URI for a given resource_type and url_param_values.
+    url_param_values should be a dict instance with keys being url parameters.
+    """
+    routes = requests.get(zc_settings.GATEWAY_ROOT_PATH).json()
+
+    for route in routes.iterkeys():
+        if 'resource_type' in routes[route] and routes[route]['resource_type'] == resource_type:
+            if len(variables(route)) != len(url_param_values):
+                raise Exception('No parameters were supplied. Required params: {0}'.format(variables(route)))
+
+            expanded = expand(route, url_param_values)
```

Comment:
It looks like this doesn't handle our arbitrary bulk get for remote resources currently. For requests like this we append a queryset of `?filter[id__in]=1,3,5` or something like that. You can reference (or probably even copy) my comment on the PR for how I did this in slots and orders.

A test case would be `get_route_from_fk('Dish', [1, 2])` should return `https://mp-vendor-services.herokuapp.com/dishes?filter[id__in]=1,2`